### PR TITLE
Add TiFlash security configuration

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

Add document for security configuration in TiFlash.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
     dt_enable_logical_split = true # 存储引擎的 segment 分裂是否使用逻辑分裂。使用逻辑分裂可以减小写放大，提高写入速度，但是会造成一定的空间浪费。默认为 true
     max_memory_usage = 0 # 单次 coprocessor 查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
     max_memory_usage_for_all_queries = 0 # 所有查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
+
+## Security settings effective since v4.0.5
```
Comment (JaySon-Huang) [inline tiflash/tiflash-configuration.md:89]:
```suggestion
## 安全相关配置，从 v4.0.5 开始生效
```

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
     dt_enable_logical_split = true # 存储引擎的 segment 分裂是否使用逻辑分裂。使用逻辑分裂可以减小写放大，提高写入速度，但是会造成一定的空间浪费。默认为 true
     max_memory_usage = 0 # 单次 coprocessor 查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
     max_memory_usage_for_all_queries = 0 # 所有查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
+
+## 安全相关配置，从 v4.0.5 开始生效
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:89]: 这里提到从 v4.0.5 开始生效，PR 是否需要 cherry-pick 到 release-4.0 呢？

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:89]: 也钩上了 pick 到 4.0。
`redact_info_log` 这个配置项暂时没有进 v4.0.9 的版本中。pick 4.0 的时候暂时先不加这个配置项。

Comment (TomShawn) [reply tiflash/tiflash-configuration.md:89]: 可以 pick 到 4.0 后再删掉 `redact_info_log`，以后再加上？

Comment (JaySon-Huang) [reply tiflash/tiflash-configuration.md:89]: 嗯，是的。

File: tiflash/tiflash-configuration.md [hunk 3]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
...
+[security]
+    ## 若开启该选项，日志中的用户数据会以 `?` 代替。如果你需要开启此选项，你需要在 tiflash-learner.toml 中
+    ## 开启 `security.redact-info-log` 以替换 tiflash-learner 日志中的用户数据。
```
Comment (flowbehappy) [inline tiflash/tiflash-configuration.md:92]:
```suggestion
    ## 若开启该选项，日志中的用户数据会以 `?` 代替。
    ## 注意，tiflash-learner 的对应选项为`security.redact-info-log`，需要另外开启。
```

**Review by flowbehappy — APPROVED** [review]
LGTM

File: tiflash/tiflash-configuration.md [hunk 4]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
     dt_enable_logical_split = true # 存储引擎的 segment 分裂是否使用逻辑分裂。使用逻辑分裂可以减小写放大，提高写入速度，但是会造成一定的空间浪费。默认为 true
     max_memory_usage = 0 # 单次 coprocessor 查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
     max_memory_usage_for_all_queries = 0 # 所有查询过程中，对中间数据的内存限制，单位为 byte，默认为 0，表示不限制
+
+## 安全相关配置，从 v4.0.5 开始生效
+[security]
+    ## 若开启该选项，日志中的用户数据会以 `?` 代替。
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:91]:
```suggestion
    ## 若开启该选项，日志中的用户数据会以 `?` 代替显示
```

File: tiflash/tiflash-configuration.md [hunk 5]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
...
+    ## 注意，tiflash-learner 的对应选项为`security.redact-info-log`，需要在 tiflash-learner.toml 中另外开启。
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:92]:
```suggestion
    ## 注意，tiflash-learner 对应的安全配置选项为 `security.redact-info-log`，需要在 tiflash-learner.toml 中另外开启
```

File: tiflash/tiflash-configuration.md [hunk 6]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
...
+    # redact_info_log = false
+
+    ## 包含可信 SSL CA 列表的文件路径。如果你设置了该值，`cert_path`, `key_path`中的路径也需要填写。
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:95]:
```suggestion
    ## 包含可信 SSL CA 列表的文件路径。如果你设置了该值，`cert_path` 和 `key_path` 中的路径也需要填写
```

File: tiflash/tiflash-configuration.md [hunk 7]
```diff
@@ -85,6 +85,19 @@ delta_index_cache_size = 0 # DeltaIndex 内存 cache 大小限制，默认为 0
...
+    # ca_path = "/path/to/ca.pem"
+    ## 包含 PEM 格式的 X509 certificate 文件路径。
+    # cert_path = "/path/to/tiflash-server.pem"
+    ## 包含 PEM 格式的 X509 key 文件路径。
+    # key_path = "/path/to/tiflash-server-key.pem"
```
Comment (TomShawn) [inline tiflash/tiflash-configuration.md:100]:
```suggestion
    ## 包含 PEM 格式的 X509 certificate 文件路径
    # cert_path = "/path/to/tiflash-server.pem"
    ## 包含 PEM 格式的 X509 key 文件路径
    # key_path = "/path/to/tiflash-server-key.pem"
```

**Review by TomShawn — COMMENTED** [review]
Minor comments on punctuation

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**